spec/lib/tasks/find_possible_duplicate_accounts_spec.rb

```diff
@@ -0,0 +1,159 @@
+  before(:all) do
+    Rake::Task["accounts:duplicate_accounts:find"] rescue Accounts::Application.load_tasks
+  end
+
+  before(:each) do
+    Rake::Task["accounts:duplicate_accounts:find"].reenable
+  end
+
+  after(:each) do
+    FileUtils.rm_f 'duplicate_users_by_name.csv'
+    FileUtils.rm_f 'duplicate_users_by_email.csv'
+  end
+
+  let(:sign_up_successful_enum) { SecurityLog.event_types[:sign_up_successful] }
+  let(:help_requested_enum) { SecurityLog.event_types[:help_requested] }
+  let(:help_request_failed_enum) { SecurityLog.event_types[:help_request_failed] }
+  let(:authentication_transfer_failed_enum) { SecurityLog.event_types[:authentication_transfer_failed] }
+
+  context "when it finds users with the same name (first name AND last name)" do
+    let!(:user_1) { FactoryGirl.create :user, first_name: "Robert", last_name: "Martin" }
+    let!(:same_name) { FactoryGirl.create :user, first_name: "robert", last_name: "martin" }
+
+    let!(:same_first_name) { FactoryGirl.create :user, first_name: user_1.first_name, last_name: "Ernser" }
+    let!(:same_last_name) { FactoryGirl.create :user, first_name: "Kaci", last_name: user_1.last_name }
+    let!(:different_name) { FactoryGirl.create :user, first_name: "Sandi", last_name: "Metz" }
+
+    before(:each) do
+      FactoryGirl.create :application_user, user: user_1
+      FactoryGirl.create :application_user, user: user_1
+
+      FactoryGirl.create :email_address, user: user_1
+      FactoryGirl.create :email_address, user: user_1, verified: true
+      FactoryGirl.create :email_address, user: same_first_name
+
+      FactoryGirl.create :security_log, event_type: :sign_up_successful, user: user_1
+      FactoryGirl.create :security_log, event_type: :help_requested, user: user_1
+      FactoryGirl.create :security_log, event_type: :help_requested, user: user_1
+      FactoryGirl.create :security_log, event_type: :help_request_failed, user: user_1
+      FactoryGirl.create :security_log, event_type: :sign_up_successful, user: same_name
+      FactoryGirl.create :security_log, event_type: :authentication_transfer_failed, user: same_name
+    end
+
+    it "creates a csv file with the results" do
+      Rake::Task["accounts:duplicate_accounts:find"].invoke
+      result = CSV.read('duplicate_users_by_name.csv', headers: true)
+
+      expect(User.count).to eq 5
+      expect(result.count).to eq 2
+      expect(SecurityLog.count).to eq 6
+
+      expect(result[0]["User First Name"]).to eq user_1.first_name
+      expect(result[0]["User Last Name"]).to eq user_1.last_name
+      expect(result[0]["Email Address(es)"]).to eq user_1.contact_infos.map{ |ci| "#{ci.value} #{ci.verified ? '(verified)' : '(NOT verified)'}" }.join(", ")
+      expect(result[0]["User ID"]).to eq user_1.id.to_s
+      expect(result[0]["Social Signup Successful?"]).to eq user_1.security_logs.where(event_type: sign_up_successful_enum).map{|sec| "On #{sec.created_at}"}.join(" and ")
```

Comment:
Also, I'd be interested in hearing what @Dantemss and @kevinburleigh75 think, but by putting the same code in this line that you have in the rake file, you're not really testing that you're getting the output that you want, only that the same code in two places returns the same results.  Ditto for the email address lines.  

E.g. you could have stored output of line 39 in `@sus` and then here check that the value equals `"On #{@sus.created_at}`.